### PR TITLE
Feat/permissions and instructions

### PR DESCRIPTION
**Pull request by kingdcreations**

### CODE REVIEW

File: android/src/main/java/com/reactnativearviewer/ArViewerView.kt [hunk 1]
```diff
@@ -54,59 +72,346 @@ open class ArViewerView @JvmOverloads constructor(
...
+    }
+    if ((context as ThemedReactContext).currentActivity != null) {
+      isStarted = true
+      try {
+        arView!!.resume()
+      } catch (ex: java.lang.Exception) {
+        sessionInitializationFailed = true
+      }
+      if (!sessionInitializationFailed) {
+        instructionsController?.setVisible(true)
```
Comment (riderodd) [inline android/src/main/java/com/reactnativearviewer/ArViewerView.kt:147]: It's already on line 131 no?

File: android/src/main/java/com/reactnativearviewer/ArViewerView.kt [hunk 2]
```diff
@@ -54,59 +72,346 @@ open class ArViewerView @JvmOverloads constructor(
...
+    // Only try once
+    if (sessionInitializationFailed) {
+      return
+    }
+    // if we have the camera permission, create the session
+    if (CameraPermissionHelper.hasCameraPermission((context as ThemedReactContext).currentActivity)) {
+      val sessionException: UnavailableException?
+      try {
+        if (requestInstall()) {
+          return
```
Comment (riderodd) [inline android/src/main/java/com/reactnativearviewer/ArViewerView.kt:170]: We should launch a JS error event

File: android/src/main/java/com/reactnativearviewer/ArViewerView.kt [hunk 3]
```diff
@@ -54,59 +72,346 @@ open class ArViewerView @JvmOverloads constructor(
...
+      updateConfig()
+    }
+  }
+
+  private fun updatePlaneDetection(config: Config?) {
+    when (planeOrientationMode) {
       "horizontal" -> {
-        nodeManipulator.setAllowedPlaneTypes(EnumSet.of(Plane.Type.HORIZONTAL_UPWARD_FACING, Plane.Type.HORIZONTAL_DOWNWARD_FACING))
-        this.planeFindingMode = Config.PlaneFindingMode.HORIZONTAL
+        config?.planeFindingMode = Config.PlaneFindingMode.HORIZONTAL
```
Comment (riderodd) [inline android/src/main/java/com/reactnativearviewer/ArViewerView.kt:332]: Is translation gesture detector limited to selected plane ?

File: android/src/main/java/com/reactnativearviewer/ArViewerView.kt [hunk 4]
```diff
@@ -54,59 +72,346 @@ open class ArViewerView @JvmOverloads constructor(
...
-        nodeManipulator.setAllowedPlaneTypes(EnumSet.allOf(Plane.Type::class.java)) // allow moving on everything
+        config?.planeFindingMode = Config.PlaneFindingMode.DISABLED
       }
     }
   }
 
-  fun onDrop() {
-    super.onDetachedFromWindow()
+  fun setInstantPlacementEnabled(isEnabled: Boolean) {
+    isInstantPlacementEnabled = isEnabled
```
Comment (riderodd) [inline android/src/main/java/com/reactnativearviewer/ArViewerView.kt:347]: Is instant placement working fine?

File: android/src/main/java/com/reactnativearviewer/ArViewerView.kt [hunk 5]
```diff
@@ -54,59 +72,346 @@ open class ArViewerView @JvmOverloads constructor(
...
+      } catch (e: UnavailableException) {
+        sessionException = e
+      } catch (e: java.lang.Exception) {
+        sessionException = UnavailableException()
+        sessionException.initCause(e)
+      }
+      sessionInitializationFailed = true
+      returnErrorEvent(sessionException?.message)
+    } else {
+      requestDangerousPermissions()
```
Comment (riderodd) [inline android/src/main/java/com/reactnativearviewer/ArViewerView.kt:194]: Aren't permissions managed on JS side? Is that still usefull ?

File: android/src/main/java/com/reactnativearviewer/ArViewerViewManager.kt [hunk 6]
```diff
@@ -31,40 +27,17 @@ class ArViewerViewManager : SimpleViewManager<ArViewerView>() {
...
   }
 
   /**
    * Pause the AR session when the view gets removed
    */
   override fun onDropViewInstance(view: ArViewerView) {
-    Log.d("ARview onDropViewInstance", "Stopping session");
+    Log.d("ARview onDropViewInstance", "Stopping session")
     super.onDropViewInstance(view)
     view.onDrop()
```
Comment (riderodd) [inline android/src/main/java/com/reactnativearviewer/ArViewerViewManager.kt:40]: Is the view unmounting without crashes?

File: android/src/main/java/com/reactnativearviewer/ArViewerView.kt [hunk 7]
```diff
@@ -54,59 +72,346 @@ open class ArViewerView @JvmOverloads constructor(
   /**
    * Set of allowed model transformations (rotate, scale, translate...)
    */
-  private var allowTransform = mutableSetOf<EditableTransform>()
+  private var allowTransform = mutableSetOf<String>()
+
+
+  init {
+    if (checkIsSupportedDevice(context.currentActivity!!)) {
```
[the remark this answers is not in the document]
Comment (kingdcreations) [reply android/src/main/java/com/reactnativearviewer/ArViewerView.kt:79]: returnErrorEvent() not showing from init(), i ll be launching it from loadModel()

File: android/src/main/java/com/reactnativearviewer/ArViewerView.kt (hunk 1 above)
Comment (kingdcreations) [reply android/src/main/java/com/reactnativearviewer/ArViewerView.kt:147]: setVisible() is different from setEnable()
If i understound well:
setEnable() activate or not the view
setVisible() display or not the image

File: android/src/main/java/com/reactnativearviewer/ArViewerView.kt (hunk 5 above)
Comment (kingdcreations) [reply android/src/main/java/com/reactnativearviewer/ArViewerView.kt:194]: Removed requestDangerousPermissions() but is CameraPermissionHelper.hasCameraPermission() still needed ?

File: android/src/main/java/com/reactnativearviewer/ArViewerViewManager.kt (hunk 6 above)
Comment (kingdcreations) [reply android/src/main/java/com/reactnativearviewer/ArViewerViewManager.kt:40]: No crashes found !

**Pull request merged**